src/main/java/org/spongepowered/common/entity/EntityUtil.java

```diff
@@ -696,11 +696,21 @@ public static Profession validateProfession(int professionId) {
                 // Vanilla behaviour - Delete the known bed location if invalid
                 bedPos = null; // null = remove location
             }
-            // Set the new bed location for the new dimension
-            int prevDim = playerIn.dimension; // Temporarily for setSpawnPoint
+
+            // First revision: Xakep_SDK, 28.01.18. MC Version: 1.12.2, API version: 7.1.0
+            // START REVISION
+            // This calls
+            // MixinEntityPlayer#onSetSpawnPoint(BlockPos pos, boolean forced, CallbackInfo ci);
```

Comment:
Where is this being called?

Reply:
Mixin magic!
https://github.com/SpongePowered/SpongeVanilla/blob/bleeding/src/main/java/org/spongepowered/server/mixin/core/entity/player/MixinEntityPlayer.java#L107